app_dart/test/request_handlers/get_branches_test.dart

```diff
@@ -38,15 +56,52 @@ void main() {
       request = FakeHttpRequest();
       keyHelper = FakeKeyHelper(applicationContext: clientContext.applicationContext);
       tester = RequestHandlerTester(request: request);
+
+      final MockGitHub github = MockGitHub();
       config = FakeConfig(
+        githubClient: github,
         dbValue: db,
         keyHelperValue: keyHelper,
       );
+
+      repositories = MockRepositoriesService();
+      when(github.repositories).thenReturn(repositories);
+
+      gerritService = FakeGerritService();
+      branchService = BranchService(
+        config: config,
+        gerritService: gerritService,
+        retryOptions: const RetryOptions(maxDelay: Duration.zero),
+      );
+
       handler = GetBranches(
+        branchService: branchService,
         config: config,
         datastoreProvider: (DatastoreDB db) => DatastoreService(config.db, 5),
       );
 
+      when(
```

Comment:
This and below can be cleaned up by ignoring unused parameters and using arrow functions

```dart
when(repositories.listBranches(Config.flutterSlug)).thenAnswer((_) async => Stream<gh.Branch>.value(gh.Branch("flutter-9.9-candidate.9", null)));
```

Reply:
umm I might have missed something, but replacing with this expression gives the following error:
`The argument type 'Future<Stream<Branch>> Function(Invocation)' can't be assigned to the parameter type 'Stream<Branch> Function(Invocation)'.`
even with `(_) async =>`

Reply:
You can remove the `async` from the arrow function.

```
when(repositories.listBranches(Config.flutterSlug)).thenAnswer((_) => Stream<gh.Branch>.value(gh.Branch("flutter-9.9-candidate.9", null)));
```

Reply:
Umm i tried removing async earlier too but it gave the error of `The argument type 'Stream<Branch> Function()' can't be assigned to the parameter type 'Stream<Branch> Function(Invocation)'`

umm just for my learning purposes, what would be the disadvantages of using the Invocation keyword?

Reply:
Synced with Casey. we use `async` for future types and remove it for non future types